tensorflow/lite/kernels/cpu_backend_context.cc

```diff
@@ -37,7 +44,7 @@ CpuBackendContext* CpuBackendContext::GetFromContext(TfLiteContext* context) {
     // We do the lazy initialization here for the TfLiteInternalBackendContext
     // that's wrapped inside ExternalCpuBackendContext.
     cpu_backend_context = new CpuBackendContext();
-    if (context->recommended_num_threads != -1) {
+    if (IsValidNumThreads(context->recommended_num_threads)) {
```

Comment:
I think the change of this value (i.e. context->recommended_num_threads) is only exposed externally via Interpreter::SetNumThreads where you already added the check. So, it looks to me adding the check here isn't that necessary.

Perhaps we could add a check to see if "recommend_num_threads" is set to -1 that indicates we will set the #(threadpool threads) to a default value (i.e. kDefaultNumThreadpoolThreads you defined here).

Reply:
@multiverse-tf : The scenario here is possibly 2 cases, where it is a fresh fetch, or re-fetch, in both the cases as it is difficult to distinguish at this layer, so it is not possible to check whether it is for default value or new value.
 Also i kept this check intentionally, i agree it is duplicate. But it is better to have 2 layer check, as both layers are not immediate next, and both are independent ones. Please let me know your valuable opinion on this. Thanks!

Reply:
Yes, I agree it's difficult to distinguish the 2 cases at this layer. But if we look outer layers of this one (note CpuBackendContext is only for TFLite runtime internal use), we could find "context->recommended_num_threads is only exposed externally via Interpreter::SetNumThreads". 

I also agree that two layers are not immediate next to each other but they are kind of tightly coupled at this moment. So, I prefer to remove the duplicate check here but I've no strong opinion here. 

The other point I mentioned in my first comment is about adding a check for value "-1", which echoes with my earlier point about the actual meaning of value "-1"  (which is to indicate using a default value set by TFLite runtime). Wdyt?

Reply:
@multiverse-tf : Thanks for the detailed reply! Its okay i think, we can remove this check.

Reply:
Acked, and Thx :-)